library/alloc/src/boxed.rs

```diff
@@ -295,6 +320,36 @@ impl<T, A: Allocator> Box<T, A> {
         unsafe { Box::from_raw_in(ptr.as_ptr(), alloc) }
     }
 
+    /// Constructs a new box with uninitialized contents in the provided allocator,
+    /// returning an error if the allocation fails
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(allocator_api, new_uninit)]
+    ///
+    /// use std::alloc::System;
+    ///
+    /// let mut five = Box::<u32, _>::try_new_uninit_in(System)?;
+    ///
+    /// let five = unsafe {
+    ///     // Deferred initialization:
+    ///     five.as_mut_ptr().write(5);
+    ///
+    ///     five.assume_init()
+    /// };
+    ///
+    /// assert_eq!(*five, 5);
+    /// # Ok::<(), std::alloc::AllocError>(())
+    /// ```
+    #[unstable(feature = "allocator_api", issue = "32838")]
+    // #[unstable(feature = "new_uninit", issue = "63291")]
+    pub fn try_new_uninit_in(alloc: A) -> Result<Box<mem::MaybeUninit<T>, A>, AllocError> {
+        let layout = Layout::new::<mem::MaybeUninit<T>>();
+        let ptr = alloc.allocate(layout)?.cast();
```

Comment:
I did not update `new_uninit_in` to be `try_new_uninit_in().unwrap_or_else()`, but can if folks prefer. I wasn't sure if it would have performance issues.

Reply:
As the current implementation of `new_uninit_in` already does an `unwrap_or_else` on failed allocations, I think it'd result in the same performance. If you want, you can still do a perf CI run to make sure though?